src/Microsoft.TestPlatform.CommunicationUtilities/SocketClient.cs

```diff
@@ -56,8 +62,28 @@ public string Start(string endPoint)
                 EqtTrace.Verbose("Waiting for connecting to server");
             }
 
-            // Don't start if the endPoint port is zero
-            this.tcpClient.ConnectAsync(ipEndPoint.Address, ipEndPoint.Port).ContinueWith(this.OnServerConnected);
+            Task connectionTask = null;
+
+            Stopwatch watch = new Stopwatch();
+            watch.Start();
+
+            // It is possible that first ConnectAsync call could not connect to server, so do a repoll to connect to server for 50secs
+            do
+            {
+                try
+                {
+                    connectionTask = this.tcpClient.ConnectAsync(ipEndPoint.Address, ipEndPoint.Port);
+                    connectionTask.Wait();
+                }
+                catch (Exception ex)
+                {
+                    EqtTrace.Verbose("Connection Failed with error {0}, retrying", ex.Message);
+                }
+            }
+            while ((this.tcpClient != null) && !this.tcpClient.Connected && watch.ElapsedMilliseconds < CONNECTIONRETRYTIMEOUT);
```

Comment:
>(this.tcpClient != null) && !this.tcpClient.Connected [](start = 19, length = 53)

suggestion: can this move into the try block.. after the wait.. and can break based on it..

Reply:
In case exception is thrown in ConnectAsync call then wait won't execute, If the connection does happen it will immediately go to while block after wait statement.
I'm not sure if we need to put the check's up, & break on basis of that.